blog/_posts/2019-08-20-multi-container-live-update.md

```diff
@@ -0,0 +1,77 @@
+---
+slug: multi-container-live-update
+date: 2019-08-20
+author: maia
+layout: blog
+title: "Live Update Multiple Containers Per Pod"
+subtitle: "Because your sidecar deserves iteration too!"
+image: sidecarDog.jpg
+image_caption: "Who WOULDN'T want to iterate on this sidecar?!"
+tags:
+  - docker
+  - kubernetes
+  - microservices
+  - tilt
+  - containers
+keywords:
+  - live-update
+  - terminal
+  - tilt
+  - development
+---
+Back in the Dark Ages (i.e. last week), the lion’s share of Tilt’s logic relied on the assumption that there was only one container per pod that was worth caring about. This was the container that we got restart counts from, that we monitored for stale code, and that we updated in place. If you were running your code on _more than one_ container, well, we could only Live Update[^1] the first one, because that was the only one we knew about. Any changes to the other containers would result in a full `docker build && kubectl apply`.
```

Comment:
I might change it to "_more than one_ container in the same pod", to reduce the chance of someone skimming this thinking it applies to simply having multiple pods

Reply:
I might also say "constraint" instead of "assumption", so that one can't read it as us naively assuming that pods only ever had one interesting container.

Similarly, something like "only one we knew about" -> "only one Tilt supported" (though even aside from the above concern, we did *know* about the other containers, we just didn't support live update on them)